src/rpm/builder.rs

```diff
@@ -148,32 +130,7 @@ impl RPMBuilder {
             arch: arch.to_string(),
             desc: desc.to_string(),
             release: "1".to_string(),
-            build_time,
-            build_host,
-            cookie: None,
-            uid: None,
-            gid: None,
-            provides: Vec::new(),
-            requires: Vec::new(),
-            obsoletes: Vec::new(),
-            conflicts: Vec::new(),
-            recommends: Vec::new(),
-            suggests: Vec::new(),
-            enhances: Vec::new(),
-            supplements: Vec::new(),
-            pre_inst_script: None,
-            post_inst_script: None,
-            pre_uninst_script: None,
-            post_uninst_script: None,
-            files: BTreeMap::new(),
-            changelog_authors: Vec::new(),
-            changelog_entries: Vec::new(),
-            changelog_times: Vec::new(),
-            compressor: Compressor::None(Vec::new()),
-            directories: BTreeSet::new(),
-            vendor: None,
-            url: None,
-            vcs: None,
+            ..Default::default()
```

Comment:
I could see a world where the replaced functionality is a feature flag. Have `hostname` as an optional package, and have a feature flag for "rpmbuild" or something, then set the values for the build host and build time conditionally if the feature flag is enabled. All it would be doing is setting the options to the calculated values.

Reply:
But why would we pull in dependencies on host functionality if we can avoid it and just expect the user to feed whatever they desire? I'd prefer to keep the feature space narrow and push things out as much as we can. Adding to many feature flags is a footgun/slowdown for development, so I'd prefer to keep it outside of this package/repo/crate. If the user wants to, they can still pull whatever dependencies in.

Reply:
If you look at the implementation, the `gethostname` dependency is only about 30 lines of code

Reply:
I always ask myself: "Is this our core concern?" "Are we adding deps that many users would possibly not desire having or prevents them from using?" And to me it's a clear 1) no - only for a vague debugging and origin tracking which is highly context specific 
2) likely no, get hostname is not supported on all platforms at least from memory

Reply:
Well it has implementations for (Li|U)nix and Windows, so that covers most platforms, but fair enough.  I understand your PoV.